src/pudl_rmi/create_override_spreadsheets.py

```diff
@@ -129,111 +150,65 @@ def _is_best_match(
+        ferc1_eia_prep = _pct_diff(ferc1_eia_prep, pct_diff_col)
 
-    # Add qualitative similarity columns (fuel_type_code_pudl)
-    check_connections.loc[
-        (check_connections.fuel_type_code_pudl_eia.notna())
-        & (check_connections.fuel_type_code_pudl_ferc1.notna()),
+    # Add in fuel_type_code_pudl diff (qualitative bool)
+    ferc1_eia_prep.loc[
+        ferc1_eia_prep.fuel_type_code_pudl_eia.notna()
+        & ferc1_eia_prep.fuel_type_code_pudl_ferc1.notna(),
         "fuel_type_code_pudl_diff",
-    ] = check_connections.fuel_type_code_pudl_eia == (
-        check_connections.fuel_type_code_pudl_ferc1
+    ] = ferc1_eia_prep.fuel_type_code_pudl_eia == (
+        ferc1_eia_prep.fuel_type_code_pudl_ferc1
     )
 
-    # Add quantitative similarity columns (installation year)
-    check_connections.loc[
+    # Add in installation_year diff (diff vs. pct_diff)
+    ferc1_eia_prep.loc[
         :, "installation_year_ferc1"
-    ] = check_connections.installation_year_ferc1.astype("Int64")
-    check_connections.loc[
-        (check_connections.installation_year_eia.notna())
-        & (check_connections.installation_year_ferc1.notna()),
+    ] = ferc1_eia_prep.installation_year_ferc1.astype("Int64")
+
+    ferc1_eia_prep.loc[
+        ferc1_eia_prep.installation_year_eia.notna()
+        & ferc1_eia_prep.installation_year_ferc1.notna(),
         "installation_year_diff",
     ] = (
-        check_connections.installation_year_eia
-        - check_connections.installation_year_ferc1
+        ferc1_eia_prep.installation_year_eia - ferc1_eia_prep.installation_year_ferc1
     )
 
-    # Move record_id_ferc1
-    record_id_ferc1 = check_connections.pop("record_id_ferc1")
-    check_connections.insert(9, "record_id_ferc1", record_id_ferc1)
-
-    # Add utility name eia
-    utils = pudl_out.utils_eia860().assign(report_year=lambda x: x.report_date.dt.year)[
-        ["utility_id_eia", "utility_name_eia", "report_year"]
-    ]
-
-    check_connections = pd.merge(
-        check_connections.dropna(
-            subset=["record_id_ferc1"]
-        ),  # dropna for now becuase somehow there is a full NA row in ferc-eia....figure out later
-        utils,
+    # Add best match col
+    ferc1_eia_prep = _is_best_match(ferc1_eia_prep)
+
+    # Add utility_name_eia
+    utils = pudl_out.utils_eia860().copy()
+    utils.loc[:, "report_year"] = utils.report_date.dt.year
+    ferc1_eia_prep = pd.merge(
+        ferc1_eia_prep,
+        utils[["utility_id_eia", "utility_name_eia", "report_year"]],
         on=["utility_id_eia", "report_year"],
         how="left",
         validate="m:1",
+        suffixes=("", "_merge"),
     )
+    # This bit is annoying -- I already have the utility_name_eia column defined and put
+    # in the correct order, but not I need to add it in again via merge. Mapping would
+    # be ideal, but it's slower. So, I'm just quickly popping the merge names and
+    # adding it to the blank utility_name_eia column.
```

Comment:
if you wanted, you could just not care about the ordering of the columns within this function and use `RENAME_COLS_FERC_EIA.values()` at the end to ensure the order at the end.

 Or you could merge `utils` in before this `for new_col in [x for x in RENAME_COLS_FERC_EIA.keys() if "new_" in x]:`

Reply:
I like the second option!!